activerecord/test/cases/database_statements_test.rb

```diff
@@ -22,6 +22,37 @@ def test_create_should_return_the_inserted_id
     assert_not_nil return_the_inserted_id(method: :create)
   end
 
+  if current_adapter?(:PostgreSQLAdapter)
+    def test_exec_insert_returns_result_with_values_for_returning_columns
+      sql = "INSERT INTO accounts (firm_id,credit_limit) VALUES (42,5000)"
+      result = @connection.exec_insert(sql, returning: ["firm_id", "credit_limit", "status"])
+      returning_values = result.first.values_at("firm_id", "credit_limit")
+      assert_equal([42, 5000], returning_values)
+    end
+
+    def test_insert_returns_values_for_returning_columns
+      sql = "INSERT INTO accounts (firm_id,credit_limit) VALUES (42,5000)"
+      result = @connection.insert(sql, returning: ["firm_id", "credit_limit", "status"])
+      assert_equal([42, 5000, nil], result)
+    end
+  else
+    def test_exec_insert_raises_when_called_with_multiple_returning_when_not_supported
+      sql = "INSERT INTO accounts (firm_id,credit_limit) VALUES (42,5000)"
+
+      assert_raises ArgumentError, match: /does not support returning more than single value from #insert/ do
+        @connection.exec_insert(sql, returning: ["firm_id", "credit_limit"])
+      end
+    end
+
+    def test_insert_raises_when_called_with_multiple_returning_when_not_supported
+      sql = "INSERT INTO accounts (firm_id,credit_limit) VALUES (42,5000)"
+
+      assert_raises ArgumentError, match: /does not support returning more than single value from #insert/ do
+        @connection.insert(sql, returning: ["firm_id", "credit_limit"])
+      end
+    end
+  end
```

Comment:
Should we add a parallel test that shows the `returning: ["id"]` case working for all adapters? Maybe even `returning: nil` to call that out?

It's obviously harder for those to be broken -- one of them definitely has to be working for _any other_ insert-related test to pass... but possible there's value in showing that _both_ are okay: the standard happy-path is mostly only going to exercise one, and even as the internal usage moves between them, they're both intended public API in their own right.